pkg/resourcemanager/apis/config/v1alpha1/types.go

```diff
@@ -251,6 +251,8 @@ type ResourceManagerWebhookConfiguration struct {
 	NodeAgentAuthorizer NodeAgentAuthorizerWebhookConfig `json:"nodeAgentAuthorizer"`
 	// SeccompProfile is the configuration for the seccomp-profile webhook.
 	SeccompProfile SeccompProfileWebhookConfig `json:"seccompProfile"`
+	// VPAInPlaceUpdates is the configuration for the vpa-in-place-or-recreate-update-mode webhook.
+	VPAInPlaceUpdates VPAInPlaceOrRecreateUpdateModeConfig `json:"vpaInPlaceOrRecreateUpdateMode"`
```

Comment:
In general, we should be consistent with the name of the field (`VPAInPlaceUpdates`) and the json key (`vpaInPlaceOrRecreateUpdateMode`).

Suggestion

1)

```suggestion
	VPAInPlaceOrRecreateUpdateMode VPAInPlaceOrRecreateUpdateModeConfig `json:"vpaInPlaceOrRecreateUpdateMode"`
```

or 

2)

```suggestion
	VPAInPlaceUpdates VPAInPlaceUpdatesConfig `json:"vpaInPlaceUpdates"`
```

If you go for, we need to adapt the GRM webhook name and other occurrences. I maybe vote for 1. - we keep the gardenlet feature gate `VPAInPlaceUpdates` but the GRM webhook stays `VPAInPlaceOrRecreateUpdateMode`. WDYT?

---

UPDATE: Looking into https://github.com/gardener/gardener/pull/12940/commits/98d12c7c1507aa928ea7cb732cf71826c5fb7ae7, I assume you renamed not only the gardenlet feature gate but also resource-manager fields. Maybe we could simply revert the resource-manager related renamings? WDYT?

Reply:
Thanks for catching this mismatch, updated the GRM `webhook` to match the feature gate's `VPAInPlaceUpdates` name, it's now `vpa-in-place-updates`. All related packages, tests and manifests are also adjusted in my latest updates.